### PR TITLE
Simple check to ignore Inline asm fwait insertion

### LINKED ISSUE #101613: Inline ASM not ignored in Insertwait pass
~$ cat k.c
```
int main(){
int val;
__asm__ volatile ("movl   %%fs:%1, %0"
:"=r" (val)
:"m" (*(int *)100)
);

}
```
~$ clang k.c -c -mno-sse -ftrapping-math
~$ objdump -d k.o

```
k.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <main>:
   0:   55                      push   %rbp
   1:   48 89 e5                mov    %rsp,%rbp
   4:   64 8b 04 25 64 00 00    mov    %fs:0x64,%eax
   b:   00
   c:   9b                      fwait
   d:   89 45 fc                mov    %eax,-0x4(%rbp)
  10:   31 c0                   xor    %eax,%eax
  12:   5d                      pop    %rbp
  13:   c3                      ret
```
As we can see thers a fwait inserted even though its not valid insertion. this was causing exception retriggering in a propreitary RTOS
IR:
`  %0 = call i32 asm sideeffect "mov   $1, $0\0Afwait", "=r,*m,~{dirflag},~{fpsr},~{flags}"(ptr elementtype(i32) inttoptr (i64 100 to ptr))
`  
  MI
`  INLINEASM &"mov   $1, $0\0Afwait" [sideeffect] [mayload] [attdialect], $0:[regdef:GR32_NOREX2], def renamable $eax, $1:[mem:m], $noreg, 1, $noreg, 100, $noreg, $2:[clobber], implicit-def dead early-clobber $df, $3:[clobber], implicit-def early-clobber $fpsw, $4:[clobber], implicit-def dead early-clobber $eflags, implicit-def $fpcw, implicit-def $mxcsr, !6`
  
  
  looks like inline asm is falsely marked as X87 instruction because of implicit-def's 
  I think inline asm need to be ignored @ https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/X86/X86InsertWait.cpp#L107

### PR DESCRIPTION
**Pull request by Temperature-block**
Just a simple check to ignore Inline asm fwait insertion

Fixes #101613

### CODE REVIEW

File: llvm/lib/Target/X86/X86InsertWait.cpp [hunk 1]
```diff
@@ -103,8 +103,9 @@ bool WaitInsert::runOnMachineFunction(MachineFunction &MF) {
 
   for (MachineBasicBlock &MBB : MF) {
     for (MachineBasicBlock::iterator MI = MBB.begin(); MI != MBB.end(); ++MI) {
-      // Jump non X87 instruction.
-      if (!X86::isX87Instruction(*MI))
+
+      // Jump non X87 instruction and Inline asm.
+      if (!X86::isX87Instruction(*MI) || MI->isInlineAsm())
```
Comment (phoebewang) [inline llvm/lib/Target/X86/X86InsertWait.cpp:108]: Put the check in `isX87Instruction`?

Comment (Temperature-block) [reply llvm/lib/Target/X86/X86InsertWait.cpp:108]: Updated, I did -ammend and force pushed. please do let me know if this is something unacceptable thanks.

Comment (Temperature-block) [reply llvm/lib/Target/X86/X86InsertWait.cpp:108]: Oops my bad  I accidentally force pushed incomplete changes, will update today sorry for any confusions.

**Comment by phoebewang** [comment]
Pre-checkin failure is relevant, please check it.

**Comment by Temperature-block** [comment]
@phoebewang  
updated with test-case change, wait needs to be inserted before final instruction in series of X87 instructions or when next instruction is a Nonwaitcontrol instruction which may reset the status registers
Please do let me know if something is amiss from my side,
Thank you

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**